content/learn/book/migration-guides/0.7-0.8/_index.md

```diff
@@ -0,0 +1,300 @@
+* a `Handle<U>` of  of a different type, consider whether this is the correct handle type to store. If it is (i.e. the same handle id is used for multiple different Asset types) use `Handle::weak(handle.id)` to cast to a different type.
+
+### [Allow higher order systems](https://github.com/bevyengine/bevy/pull/4833)
+
+<!-- TODO marked as breaking change, but need to figure out what actually is breaking -->
+
+### [Added offset parameter to TextureAtlas::from_grid_with_padding](https://github.com/bevyengine/bevy/pull/4836)
+
+Calls to `TextureAtlas::from_grid_with_padding` should be modified to include a new parameter, which can be set to `Vec2::ZERO` to retain old behaviour.
+
+```rust
+from_grid_with_padding(texture, tile_size, columns, rows, padding)
+                                  |
+                                  V
+from_grid_with_padding(texture, tile_size, columns, rows, padding, Vec2::ZERO)
+```
+
+### [Split mesh shader files](https://github.com/bevyengine/bevy/pull/4867)
+
+* In shaders for 3D meshes:
+  * `#import bevy_pbr::mesh_view_bind_group` -> `#import bevy_pbr::mesh_view_bindings`
+  * `#import bevy_pbr::mesh_struct` -> `#import bevy_pbr::mesh_types`
+    * NOTE: If you are using the mesh bind group at bind group index 2, you can remove those binding statements in your shader and just use `#import bevy_pbr::mesh_bindings` which itself imports the mesh types needed for the bindings.
+* In shaders for 2D meshes:
+  * `#import bevy_sprite::mesh2d_view_bind_group` -> `#import bevy_sprite::mesh2d_view_bindings`
+  * `#import bevy_sprite::mesh2d_struct` -> `#import bevy_sprite::mesh2d_types`
+    * NOTE: If you are using the mesh2d bind group at bind group index 2, you can remove those binding statements in your shader and just use `#import bevy_sprite::mesh2d_bindings` which itself imports the mesh2d types needed for the bindings.
+
+### [Camera Driven Viewports](https://github.com/bevyengine/bevy/pull/4898)
+
+`Camera::projection_matrix` is no longer a public field. Use the new `Camera::projection_matrix()` method instead:
+
+```rust
+
+// 0.7
+let projection = camera.projection_matrix;
+
+// 0.8
+let projection = camera.projection_matrix();
+```
+
+### [diagnostics: meaningful error when graph node has wrong number of inputs](https://github.com/bevyengine/bevy/pull/4924)
+
+Exhaustive matches on `RenderGraphRunnerError` will need to add a branch to handle the new `MismatchedInputCount` variant.
+
+### [Make Reflect safe to implement](https://github.com/bevyengine/bevy/pull/5010)
+
+* Reflect derives should not have to change anything
+* Manual reflect impls will need to remove the `unsafe` keyword, add `any()` implementations, and rename the old `any` and `any_mut` to `as_any` and `as_mut_any`.
+* Calls to `any`/`any_mut` must be changed to `as_any`/`as_mut_any`
+
+### [Mark mutable APIs under ECS storage as pub(crate)](https://github.com/bevyengine/bevy/pull/5065)
+
+<!-- TODO not sure what to say here -->
```

Comment:
If any user cared, they should file an issue explaining what on earth they were doing.